drivers/scheduler/k8s/k8s.go

```diff
@@ -2842,13 +2842,23 @@ func (k *K8s) ScaleApplication(ctx *scheduler.Context, scaleFactorMap map[string
 		}
 		if obj, ok := specObj.(*appsapi.Deployment); ok {
 			logrus.Infof("Scale all Deployments")
-			dep, err := k8sOps.GetDeployment(obj.Name, obj.Namespace)
-			if err != nil {
-				return err
-			}
 			newScaleFactor := scaleFactorMap[obj.Name+DeploymentSuffix]
-			*dep.Spec.Replicas = newScaleFactor
-			if _, err := k8sOps.UpdateDeployment(dep); err != nil {
+
+			t := func() (interface{}, bool, error) {
+				dep, err := k8sOps.GetDeployment(obj.Name, obj.Namespace)
+				if err != nil {
+					return "", false, err // do not retry if we fail to get dep
```

Comment:
why not retry this as well?

Reply:
didn't see this error in the test, so not sure if it is worth retrying. I could retry since you ask